src/Equinox.Core/Category.fs

```diff
@@ -32,41 +32,39 @@ open System.Threading.Tasks
+                if act <> null then act.AddStream(name, streamId, streamName).AddLeader(requireLeader).AddStale(maxAge) |> ignore
+                return! inner.Load(log, name, streamId, streamName, maxAge, requireLeader, ct) }
+            member _.Sync(attempt, token, originState, events, ct) = task {
+                use act = source.StartActivity("Sync", ActivityKind.Client)
+                if act <> null then act.AddStream(name, streamId, streamName).AddSyncAttempt(attempt) |> ignore
                 let log = if attempt = 1 then log else log.ForContext("attempts", attempt)
-                return! inner.TrySync(log, categoryName, streamId, streamName, context, init, token, originState, events, ct) } }
-
-[<AbstractClass; Sealed>]
-type private Stream private () =
-    static member Resolve(cat: Category<'event, 'state, 'context>, log, context): System.Func<string, Core.StreamId, Core.IStream<'event, 'state>> =
-        System.Func<string, Core.StreamId, _>(fun categoryName streamId -> cat.Stream(log, context, categoryName, Core.StreamId.toString streamId))
+                return! inner.Sync(log, name, streamId, streamName, context, init, token, originState, events, ct) } }
 
 [<AbstractClass; Sealed; System.Runtime.CompilerServices.Extension>]
-type Factory private () =
+type Stream private () =
     [<System.Runtime.CompilerServices.Extension>]
-    static member Resolve(cat: Category<'event, 'state, 'context>, log, context): System.Func<string, Core.StreamId, DeciderCore<'event, 'state>> =
-         System.Func<_, _, _>(fun c s -> Stream.Resolve(cat, log, context).Invoke(c, s) |> DeciderCore<'event, 'state>)
+    static member Resolve(cat: Category<'event, 'state, 'context>, log, context): System.Func<Core.StreamId, DeciderCore<'event, 'state>> =
+         System.Func<_, _>(fun sid -> cat.Stream(log, context, Core.StreamId.toString sid) |> DeciderCore<'event, 'state>)
     [<System.Runtime.CompilerServices.Extension>]
-    static member Resolve(cat: Category<'event, 'state, unit>, log): System.Func<string, Core.StreamId, DeciderCore<'event, 'state>> =
-        Factory.Resolve(cat, log, ())
+    static member Resolve(cat: Category<'event, 'state, unit>, log): System.Func<Core.StreamId, DeciderCore<'event, 'state>> =
+        Stream.Resolve(cat, log, ())
 
 module Decider =
-    let resolveWithContext log (cat: Category<'event, 'state, 'context>) categoryName context streamId =
-        Factory.Resolve(cat, log, context).Invoke(categoryName, streamId) |> Decider<'event, 'state>
-    let resolve log (cat: Category<'event, 'state, unit>) categoryName streamId =
-        resolveWithContext log cat categoryName () streamId
+    let forContext log (cat: Category<'event, 'state, 'context>) context streamId =
+        Stream.Resolve(cat, log, context).Invoke(streamId) |> Decider<'event, 'state>
+    let forStream log (cat: Category<'event, 'state, unit>) streamId =
```

Comment:
I think `for` is better 🤔

Reply:
Enough better to use `For` ? :P

Reply:
No 🙅 

what's wrong with `resolveStream`?

Reply:
sorry, scanned that too quick - are you suggesting `resolveStream` instead of `forStream`?
While there is a resolution of cat+streamId to container/table + streamName, this just uses the `Stream` on the Cat, making a `Decider` based on that stream?

Reply:
🤔 , yeah, I think `forStream` is the way to go.

`Decider.forStream yuckyLog cat streamId` reads well

Reply:
> Which, depending on the arg order of ctx vs sid is either (right now in this PR as it stands)

```ts
Equinox.Decider.forContext Metrics.log cat
```

Weirdly enough, this is what it looks like in my code, because in my codebase we always pass context first

Reply:
>Weirdly enough, this is what it looks like in my code, because in my codebase we always pass context first
So what's the full sugar, where do you add the `streamId` mapping?